### PR TITLE
Handle `Set-Cookie` in `str` correctly

### LINKED ISSUE #298: `str` function incorrectly implodes all headers
Headers that cannot carry multiple values need to be output as separate lines, most annoying example is `Set-Cookie` so if the Response carries multiple values for it `str` will incorrectly add them comma separated to one `Set-Cookie` header.

Solution would be to check for such (known "bad") headers and output them as separate lines.

### PR DESCRIPTION
**Pull request by albe**
Fixes #298

### CODE REVIEW

**Comment by albe** [comment]
Maybe there are some other headers that fall into this special handling, in which case the list of special header names should probably end up in a static array list. Needs some investigation.

**Comment by GrahamCampbell** [comment]
> Maybe there are some other headers that fall into this special handling, in which case the list of special header names should probably end up in a static array list. Needs some investigation.

Maybe, but this PR certainly fixes the most common case, and the bug report, so 👍  from me.

**Comment by GrahamCampbell** [comment]
Just needs some tests, then it'd be good to merge IMO.

**Comment by albe** [comment]
Completely lost track of this one... will try to tackle the missing tests ASAP, as it's not too hard really... feel free to nag me if I'm not doing it within a week.

**Comment by Tobion** [comment]
@albe I'm nagging you as you asked for it :)

**Comment by albe** [comment]
I also took a look if I could find any other header that behaves like `Set-Cookie` in that regard, but could not find anything. Basically all other headers standardized only make sense when sent only once.
One example of another "multi-value header" would maybe be `Link`, but that encodes the multiple values with semicolon-separated values. The reason `Set-Cookie` doesn't work like that, is that each single cookie itself is a multi-value type, which encodes the values with semicolon separation.

Fun fact: The old RFC actually specified that multiple cookies be separated by comma in a single Set-Cookie header (https://tools.ietf.org/html/rfc2109#section-4.2.2). That RFC has been superseeded though and the current version only allows single cookie per header (rfc6265)

References:
https://en.wikipedia.org/wiki/List_of_HTTP_header_fields
https://tools.ietf.org/html/rfc6265#section-4.1.1
https://tools.ietf.org/html/rfc5988#section-5


tl;dr: Most likely `Set-Cookie` really is a single outlier

**Comment by hlubek** [comment]
> I also took a look if I could find any other header that behaves like `Set-Cookie` in that regard, but could not find anything. Basically all other headers standardized only make sense when sent only once.

Have a look at https://book.varnish-software.com/4.0/chapters/Cache_Invalidation.html#example-using-hashtwo-or-xkey, this is an example for a header requiring multiple instances.
